Source/Core/Core/NetPlayClient.cpp

```diff
@@ -895,6 +1121,122 @@ bool NetPlayClient::StartGame(const std::string& path)
   return true;
 }
 
+void NetPlayClient::SyncSaveDataResponse(const bool success)
+{
+  m_dialog->AppendChat(success ? GetStringT("Data received!") :
+                                 GetStringT("Error processing data."));
+
+  if (success)
+  {
+    if (++m_sync_save_data_success_count >= m_sync_save_data_count)
+    {
+      sf::Packet response_packet;
+      response_packet << static_cast<MessageId>(NP_MSG_SYNC_SAVE_DATA);
+      response_packet << static_cast<MessageId>(SYNC_SAVE_DATA_SUCCESS);
+
+      Send(response_packet);
+    }
+  }
+  else
+  {
+    sf::Packet response_packet;
+    response_packet << static_cast<MessageId>(NP_MSG_SYNC_SAVE_DATA);
+    response_packet << static_cast<MessageId>(SYNC_SAVE_DATA_FAILURE);
+
+    Send(response_packet);
+  }
+}
+
+bool NetPlayClient::DecompressPacketIntoFile(sf::Packet& packet, const std::string& file_path)
+{
+  u32 size_low, size_high;
+  packet >> size_low >> size_high;
+  u64 file_size = size_low | (static_cast<u64>(size_high) << 32);
+
+  if (file_size == 0)
+    return true;
+
+  File::IOFile file(file_path, "wb");
+  if (!file)
+  {
+    PanicAlertT("Failed to open file \"%s\". Verify your write permissions.", file_path.c_str());
+    return false;
+  }
+
+  std::vector<u8> in_buffer(NETPLAY_LZO_OUT_LEN);
+  std::vector<u8> out_buffer(NETPLAY_LZO_IN_LEN);
+
+  while (true)
+  {
+    lzo_uint32 cur_len = 0;  // number of bytes to read
+    lzo_uint new_len = 0;    // number of bytes to write
+
+    packet >> cur_len;
+    if (!cur_len)
+      break;  // We reached the end of the data stream
+
+    for (size_t j = 0; j < cur_len; j++)
+    {
+      packet >> in_buffer[j];
+    }
+
+    if (lzo1x_decompress(in_buffer.data(), cur_len, out_buffer.data(), &new_len, nullptr) !=
+        LZO_E_OK)
+    {
+      PanicAlertT("Internal LZO Error - decompression failed");
+      return false;
+    }
+
+    if (!file.WriteBytes(out_buffer.data(), new_len))
+    {
+      PanicAlertT("Error writing file: %s", file_path.c_str());
+      return false;
+    }
+  }
+
+  return true;
+}
+
+std::optional<std::vector<u8>> NetPlayClient::DecompressPacketIntoBuffer(sf::Packet& packet)
```

Comment:
This method looks awfully similar to `DecompressPacketIntoFile`, I wonder if you could implement one in terms of the other (or use a 3rd shared method with common code instead).

Reply:
I wanted `DecompressPacketIntoFile` to write directory to the file rather than buffering all the data first, which is why it has to have similar duplicated code.

Reply:
I understand.
It would still be possible to do this, but I'm not sure if its worth it (since it might turn out to be a lot more involved than just copying the method and replacing a few lines). Ok to ignore this comment (unless someone else has a cheap idea on how to get rid of the duplication)